hnn_core/network_builder.py

```diff
@@ -497,65 +500,88 @@ def _parnet_connect(self):
         # layer5 Pyr -> layer5 Pyr
         nc_dict['lamtha'] = 3.
         for target_cell in ['L2Pyr', 'L5Pyr']:
+            target_gids = self.net.gid_ranges[_long_name(target_cell)]
             for receptor in ['nmda', 'ampa']:
                 key = f'gbar_{target_cell}_{target_cell}_{receptor}'
                 nc_dict['A_weight'] = params[key]
-                self._connect_celltypes(target_cell, target_cell, 'proximal',
+                self._connect_celltypes(target_gids, target_gids, 'proximal',
                                         receptor, nc_dict,
                                         allow_autapses=False)
 
         # layer2 Basket -> layer2 Pyr
+        src_cell = 'L2Basket'
+        src_gids = self.net.gid_ranges[_long_name(src_cell)]
```

Comment:
this is a cool idea :) Let's see how far you get!

Reply:
I agree! Great stuff :)